public/app/features/dashboard/dashgrid/PanelPerformanceMonitor.tsx

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useRef } from 'react';
+
+import { FieldConfigSource } from '@grafana/data';
+import { faro } from '@grafana/faro-web-sdk';
+import appEvents from 'app/core/app_events';
+import { DashboardSavedEvent } from 'app/types/events';
+
+interface Props {
+  children: JSX.Element;
+  isInPanelEdit: boolean;
+  panelType: string;
+  panelId: number;
+  panelTitle: string;
+  panelOptions: any;
+  panelFieldConfig: FieldConfigSource;
+}
+
+export const PanelPerformanceMonitor = (props: Props) => {
+  const startLoadTime = performance.now();
+  const panelIdStr = String(props.panelId);
+
+  const oldPanelOptions = useRef(props.panelOptions);
+  const oldPanelFieldConfig = useRef<FieldConfigSource>(props.panelFieldConfig);
+
+  const savedFieldConfig = useRef<FieldConfigSource>(props.panelFieldConfig);
+  const savedPanelOptions = useRef(props.panelOptions);
+
+  const logChanges = () => {
+    logPanelOptionChanges(savedPanelOptions.current, oldPanelOptions.current);
+    logFieldConfigChanges(savedFieldConfig.current, oldPanelFieldConfig.current);
+
```

Comment:
these are better logged when a user makes actual changes, rather then on render

Reply:
and maybe on mount (or some time after mount to ont impact load dashboard speed)

Reply:
The option logging would only happen in two situations: if the user is in panel edit page and saves the dashboard, or if the user leaves the panel edit page, because we lose the props for the diff in the dashboard page and might potentially not want to miss their (yet) unsaved changes